src/darwin/Framework/CHIP/CHIPDevice.mm

```diff
@@ -1109,7 +1061,7 @@ - (void)subscribeAttributeWithEndpointId:(NSUInteger)endpointId
         auto callback = chip::Platform::MakeUnique<BufferedReadAttributeCallback<NSObjectData>>(
             chipClusterId, chipAttributeId, onReportCb, onFailureCb, onDone, onEstablishedCb);
 
-        auto readClient = new app::ReadClient(engine, [self internalDevice]->GetExchangeManager(),
+        auto readClient = Platform::New<app::ReadClient>(engine, [self internalDevice]->GetExchangeManager(),
```

Comment:
Don't the `container.pathParams` also need this change?  How are they deleted?

Reply:
Good catch. They are deleted from container dealloc method with delete. Both new and delete must be replaced with Platform::New and Platform::Delete to be consistent.